src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1741,6 +1750,17 @@ private static SyntaxTokenList AsModifierList(Accessibility accessibility, Decla
         if (modifiers.IsPartial)
             list.Add(PartialKeyword);
 
+        for (int i = 0, n = list.Count; i < n; i++)
+        {
+            // By default, do not place leading elastic trivia on modifiers we make.  Just because something is
+            // adding/removing/modifying modifiers does not mean we want the parent construct to change its formatting
+            // respective to what's around it.
+            if (!withLeadingElasticMarker)
+                list[i] = list[i].WithoutLeadingTrivia();
```

Comment:
the crux of the fix.  we do *not* want elastic trivia at teh start of a member by default, and especially not if we're just mutating modifiers.  We want the modifiers to just be inertly placed into the member, leaving the formatting the same as how it started.  

callers can choose if they do want leading elastic trivia added